cmd/generate-workflow.go

```diff
@@ -1,27 +1,36 @@
 package cmd
 
 import (
+	"github.com/Azure/draft/pkg/templatewriter"
+	"github.com/Azure/draft/pkg/templatewriter/writers"
 	"github.com/Azure/draft/pkg/workflows"
 	"github.com/spf13/cobra"
 
 	log "github.com/sirupsen/logrus"
 )
 
+type generateWorkflowCmd struct {
+	workflowConfig workflows.WorkflowConfig
+	dest           string
+	flagVariables  []string
+	templateWriter templatewriter.TemplateWriter
+}
+
 func newGenerateWorkflowCmd() *cobra.Command {
 
-	workflowConfig := &workflows.WorkflowConfig{}
-	dest := ""
+	gwCmd := &generateWorkflowCmd{}
+	gwCmd.dest = ""
 	var cmd = &cobra.Command{
 		Use:   "generate-workflow [flags]",
 		Short: "Generates a Github workflow for automatic build and deploy to AKS",
 		Long: `This command will generate a Github workflow to build and deploy an application containerized 
 with draft on AKS. This command assumes the 'setup-gh' command has been run properly.`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			workflowConfig.ValidateAndFillConfig()
+			gwCmd.workflowConfig.ValidateAndFillConfig()
```

Comment:
it looks like this would still prompt the user for all the inputs even if they are supplied in flags since ValidateAndFillConfig() runs before anything else